HIRS_ProvisionerTPM2/CMakeLists.txt

```diff
@@ -234,7 +234,7 @@ if(STATIC_ANALYSIS)
             --library=posix.cfg
             --error-exitcode=1
             --verbose
-            -i src/ProvisionerTpm2.pb.cc
+            --suppress=readdirCalled
```

Comment:
What is this change for?

Reply:
Yeah so in the other implementation where we doing what `pgrep` did, it relied on a method called `readdir`, which isn't thread safe, but they deprecated the version of that was thread safe called `readdir_r` so we had to use `readdir`. CppCheck didn't appreciate that so I had to suppress the warning to get the build to pass.

The strange part is `readdir_r` was even more unsafe than the non-thread safe version, thus the deprecation, so I figured if anyone ever needed to use the function in the future they would need to use this suppression. I'm not sure if the error would go away if it was surrounded by a `mutex::lock` but the error message didn't make that suggestion at the time I was tinkering with it.